src/libraries/System.Runtime.InteropServices.JavaScript/src/System/Runtime/InteropServices/JavaScript/SynchronizationContextExtensions.cs

```diff
@@ -61,6 +61,86 @@ public static TRes Send<TRes>(this SynchronizationContext? self, Func<TRes> body
             return value!;
         }
 
+        public static Task<TRes> Post<TRes>(this SynchronizationContext? self, Func<Task<TRes>> body)
+        {
+            if (self == null) return body();
```

Comment:
Is it possible this will hide bugs?

Reply:
This whole extension class will go away in one of the next PRs when I move the logic to inside of `JSImport` guts.

The idea here was that `null` `SynchronizationContext` means same as `new SynchronizationContext()` that's the default. 
Because `SynchronizationContext.Current` is nullable and could be null.
But you are right, I will assert instead. Especially because in this signature, it always needs to be scheduled, not inlined.